idea-plugin/src/main/kotlin/arrow/meta/ide/plugins/proofs/inspections/explicitcoercion.kt

```diff
@@ -0,0 +1,102 @@
+package arrow.meta.ide.plugins.proofs.inspections
+
+import arrow.meta.ide.IdeMetaPlugin
+import arrow.meta.ide.dsl.IdeSyntax
+import arrow.meta.ide.plugins.proofs.markers.participatingTypes
+import arrow.meta.phases.CompilerContext
+import arrow.meta.phases.ExtensionPhase
+import arrow.meta.plugins.proofs.phases.areTypesCoerced
+import arrow.meta.plugins.proofs.phases.coerceProof
+import arrow.meta.quotes.ktFile
+import com.intellij.codeHighlighting.HighlightDisplayLevel
+import com.intellij.codeInspection.ProblemHighlightType
+import com.intellij.psi.PsiElement
+import org.jetbrains.kotlin.idea.imports.importableFqName
+import org.jetbrains.kotlin.idea.inspections.AbstractApplicabilityBasedInspection
+import org.jetbrains.kotlin.psi.KtElement
+import org.jetbrains.kotlin.psi.KtExpression
+import org.jetbrains.kotlin.psi.KtImportDirective
+import org.jetbrains.kotlin.psi.KtProperty
+import org.jetbrains.kotlin.psi.KtValueArgument
+import org.jetbrains.kotlin.resolve.ImportPath
+import org.jetbrains.kotlin.types.KotlinType
+import org.jetbrains.kotlin.utils.addToStdlib.safeAs
+
+/**
+ * [explicitCoercionInspection]: for implicit coercion to make it explicit
+ */
+val IdeMetaPlugin.explicitCoercionInspection: ExtensionPhase
+  get() = addLocalInspection(
+    inspection = explicitCoercionInspectionSyntax,
+    level = HighlightDisplayLevel.WEAK_WARNING,
+    groupPath = ProofPath + arrayOf("Coercion")
+  )
+
+const val EXPLICIT_COERCION_INSPECTION_ID = "Make_coercion_explicit"
+
+val IdeSyntax.explicitCoercionInspectionSyntax: AbstractApplicabilityBasedInspection<KtElement>
+  get() = applicableInspection(
+    defaultFixText = EXPLICIT_COERCION_INSPECTION_ID,
+    inspectionHighlightType = { ProblemHighlightType.INFORMATION },
+    kClass = KtElement::class.java,
```

Comment:
Ideally, this is split up in 2 different inspections. We would split this feature based on the ast node it matches on

Reply:
Isn't that what's happening already for `explicitcoercion.kt` vs `implicitcoercion.kt`?

Reply:
noo, @i-walker means.. instead of targeting the KtElement and then work with both KtValueArgument and KtProperty split this.. and target independently these two.. right?